src/Symfony/Component/HttpFoundation/Tests/ResponseTest.php

```diff
@@ -211,6 +211,29 @@ public function testIsNotModifiedEtag()
         $this->assertFalse($response->isNotModified($request));
     }
 
+    public function testIsNotModifiedEtagWithResponseSetter()
+    {
+        $etag = 'random_etag';
+        $weakEtag = 'weak_etag';
+
+        $request = new Request();
+        $request->headers->set('if_none_match', sprintf('\\"%s\\", W/\\"%s\\"', $etag, $weakEtag));
```

Comment:
This test case is strange. Headers should already been escaped. 
If this is not the case, the fix should not be about this specific "if_none_match" header, but everything in the request.

note: The original issue  #38815 is not about unescapped quotes, but about etag `foo` not matching `"foo"`

Reply:
As described [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match), the value of the ETag is always quoted and may be optional prefixed with `W/`. If the HTTP request has header `If-None-Match:"foo"`, the value of `$request->getEtags()` is `\"foo\"`, not `"foo"`. I checked both `$_SERVER` global and `Request` class, all values of the `If-None-Match` header is escaped there. This is the reason for the test case.

Furthermore, the reason for the issue described in #38815 is that when we set the ETag to the response object, the getter always return an unescaped quoted value (here, `"foo"`) whereas the request only returns an escaped quoted value (`\"foo\"`) or unquoted value (`foo`).

I actually though of two approaches:
* Update the `Response` class as in this MR, which does not introduce BC
* Update the `Request` class to return the escaped value of the `If-None-Match` header. I think it will introduce a BC, right?

I think the changes in the `Response` class is less expensive.